docs/source/concepts/ledger-model/ledger-exceptions.rst

```diff
@@ -122,6 +122,84 @@ And the third:
 
 As you can see, in each of these continuities, no contract was consumed twice.
 
+Transaction Normalization
++++++++++++++++++++++++++
+
+The same "before-after" relation can be represented in more than one way using
+rollback nodes. For example, the following three transactions have the same
+"before-after" relation among their ledger actions (A and B):
+
+.. https://lucid.app/lucidchart/3aa5922f-ec30-4896-8bbc-56703549c7e5/edit
+.. image:: ./images/exception-normalization-1.svg
+   :align: center
+   :width: 80%
+
+Because of this, these three transactions are equivalent.
+More generally, two transactions are equivalent if:
+
+- The transactions are the same when you ignore all rollback nodes. That is,
+  if you remove every rollback node and absorb its children into its parent,
+  then two transactions are the same. Equivalently, the transactions have
+  the same ledger actions with the same prefix order and subaction relation.
+
+- The transactions have the same "before-after" relation between their actions.
+
+- The trensactions have the same set of rolled back actions. That is, they have
+  the same set of actions directly or indirectly under a rollback node.
+
+For all three transactions above, the "transaction tree ignoring rollbacks"
+consists of just the top-level actions A and B, the actions A and B are not
+related by the "before-after" relation, and both A and B are rolled back.
+Thus all three transactions are equivalent.
+
+**Transaction normalization** is the process by which equivalent transactions
+are converted into the same transaction. In the case above, all three
+transactions become the transaction in the middle when normalized.
+
+.. https://lucid.app/lucidchart/8a5a09a1-5473-4abf-a72a-57bf03b56794/edit
+.. image:: ./images/exception-normalization-2.svg
+   :align: center
+   :width: 80%
+
+To normalize a transaction, we apply three rules across the whole transaction:
```

Comment:
Minor: Do you want to walk the reader through the normalization of the diagrams? It's not plain obvious how to get from the left to the middle (and there are actually two ways to get there: use "pull out the start" or "pull out the end" and then remove the empty remaining rollback node.

Reply:
Good idea. I changed the example a bit to eliminate ambiguity and to showcase both rules 2 and 3.